packages/target-tauri/src-tauri/src/main.rs

```diff
@@ -2,5 +2,11 @@
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
 fn main() {
-    deltachat_tauri_lib::run()
+    let exit_code = deltachat_tauri_lib::run();
+    if exit_code != 0 {
+        // We could probably just run this regardless of the code,
+        // but let's be on the safe side and ensure
+        // that all `drop()`s are run on normal exit.
+        std::process::exit(exit_code);
+    }
```

Comment:
as far as I understand rust drop was already run at this point.

we could also return an exit code: https://doc.rust-lang.org/std/process/struct.ExitCode.html